pkg/api/dataproxy_test.go

```diff
@@ -7,28 +7,55 @@ import (
 )
 
 func TestDataProxy(t *testing.T) {
-	testCases := []struct {
-		desc      string
-		origPath  string
-		proxyPath string
-		exp       string
-	}{
-		{
-			"Should append trailing slash to proxy path if original path has a trailing slash",
-			"/api/datasources/proxy/6/api/v1/query_range/",
-			"api/v1/query_range/",
-			"api/v1/query_range/",
-		},
-		{
-			"Should not append trailing slash to proxy path if original path doesn't have a trailing slash",
-			"/api/datasources/proxy/6/api/v1/query_range",
-			"api/v1/query_range",
-			"api/v1/query_range",
-		},
-	}
-	for _, tc := range testCases {
-		t.Run(tc.desc, func(t *testing.T) {
-			assert.Equal(t, tc.exp, ensureProxyPathTrailingSlash(tc.origPath, tc.proxyPath))
-		})
-	}
+	t.Run("ensureProxyPathTrailingSlash", func(t *testing.T) {
+		testCases := []struct {
+			desc      string
+			origPath  string
+			proxyPath string
+			exp       string
+		}{
+			{
+				"Should append trailing slash to proxy path if original path has a trailing slash",
+				"/api/datasources/proxy/6/api/v1/query_range/",
+				"api/v1/query_range/",
+				"api/v1/query_range/",
+			},
+			{
+				"Should not append trailing slash to proxy path if original path doesn't have a trailing slash",
+				"/api/datasources/proxy/6/api/v1/query_range",
+				"api/v1/query_range",
+				"api/v1/query_range",
+			},
+		}
+		for _, tc := range testCases {
+			t.Run(tc.desc, func(t *testing.T) {
+				assert.Equal(t, tc.exp, ensureProxyPathTrailingSlash(tc.origPath, tc.proxyPath))
+			})
+		}
+	})
+
+	t.Run("getProxyPath", func(t *testing.T) {
+		testCases := []struct {
+			originalRawPath string
+			exp             string
+		}{
+			{
+				"/api/datasources/proxy/1",
+				"",
+			},
+			{
+				"/api/datasources/proxy/1/some/thing",
+				"some/thing",
+			},
+			{
+				"/api/datasources/proxy/54/api/services/afsd%2Fafsd/operations",
+				"api/services/afsd%2Fafsd/operations",
+			},
+		}
+		for _, tc := range testCases {
+			t.Run("Given raw path should equal expected proxy path", func(t *testing.T) {
```

Comment:
I struggle with making sense of the original description? Does the following make sense?
```suggestion
			t.Run("Given raw path, should produce expected proxy path", func(t *testing.T) {
```